### ISSUE TITLE
Subsegment not emited after first lambda invocation

### ISSUE THREAD

**Issue by jeremielec** [aws/aws-xray-sdk-dotnet#240]
I have an issue with the XRay SDK for .Net.

Context : Hosting an AspNet Core 6.0 Api within custom runtime. Lambda are invoked throught Api gateway.

Any subsequent segment created after the first invocation are just lost and not displayed therefore in XRay console.

Note that the EF Core interceptor are used for an SQL Server database. Query are visible on the first invocation, not the subsequent.

Attached log file.

=> On first invocation, Subsegment emit log are cleary visible.

=> On subsequent, no more log, and there are not visible at all in console

[log.txt](https://github.com/aws/aws-xray-sdk-dotnet/files/8390373/log.txt)

Thank in advance for any help for troubleshooting that  strange issue

**Comment by jon-armen** [comment]
I do not believe the current EF Core Interceptor will work on dotnet6.0.  I have several branches and plan to implement the fixes required once this library starts releasing more frequently and makes progress on the 3.x breaking changes release.

**Comment by srprash** [comment]
@jon-armen could be right about EFCore Interceptor not functional with .Net6 but I suspect if that is the main root cause here since @jeremielec said none of the segments are displayed.

@jeremielec Can you provide screenshots of the trace with all the segments and subsegments present, and the one with issues? Also, within your Lambda function, are you generating additional segments or subsegments other than the one for SQL server db?

**Comment by jon-armen** [comment]
@jeremielec , I'm also curious, are your calls to EF sync or async?  Reviewing the changes for net6.0, sync may work fine.  We would likely see a missing method exception for any async method as the return types changed from Task to ValueTask

**Comment by jeremielec** [comment]
@jon-armen  All EfCore call are done in async patern.

How over there is no missing method exception at all, everything is just working fine.

@srprash  No additional segment / subsegment are created.
Since this is an internal application, i have hidden sensitive data (on promise SQL server host, lambda name)

Xray trace valid :

![image](https://user-images.githubusercontent.com/46441464/164192678-720ed284-2351-4dbb-ae5e-1d64d2ac90ce.png)

XRay with missing data :

![image](https://user-images.githubusercontent.com/46441464/164192703-ae301acf-eb03-4fab-ba0d-71519133852d.png)

Api call has been done within 10 second interval with postman.

From lambda log stream, there is clear outgoing sql server request executed.

**Comment by srprash** [comment]
Hey @jeremielec 
Thanks for the screenshot. Looking at them I couldn't tell why there are no subsegments for the DB call. However, I noticed that there is a `Database::SQL` node from the `AWS::Lambda::Function` in the screenshot, which means that the request to database is being captured but not being shown. Probably a bug in the console rendering?
<img width="1180" alt="Screen Shot 2022-04-24 at 2 41 30 PM" src="https://user-images.githubusercontent.com/50466688/164997920-021c5c5a-8f0a-4f8d-bfd4-e4f3da5a2b90.png">


Can you check the raw data for the trace and verify if the `Database::SQL` subsegments are present there? And if possible provide a snippet of it. Thanks.

**Comment by jeremielec** [comment]
@srprash   Hello

No SQL data are visible at all on Raw Data tab.  These one are visible on the (good ?) trace with full sql information.

Since no information to view, i cannot provided a snippet :/ 

How over regarding the display problem mentionned, there is a bug in XRay console after changing the XRay trace id, External node service are updated after a couple of second without any notification. Image of the problem are more relevant :

Immediatly after copy paste the trace which has missing subsegment problem :

![image](https://user-images.githubusercontent.com/46441464/165096656-e8b163b0-d92d-400a-9449-5d87db42d316.png)

After a couple of second : (the screen blink when the "orphaned" node disappear) 

![image](https://user-images.githubusercontent.com/46441464/165096836-bc664515-30b0-498f-8132-8c90452b2e83.png)
